### PR TITLE
Fix missing prefix 'v' for versions v0.41.0 and v0.42.0

### PR DESCRIPTION
**Pull request by rhrazdil**
1. commit __Fixed__ missing `v` prefix in released manifest container tags in versions v0.41.0 and v0.42.0.

2. commit __Updated__ `prepare-release.sh` to generated manifests with the prefixed CONTAINER_TAG value.

From version v0.41.0 forward, we are using v-prefixed versions in **git tags** and **CONTAINER_TAG**s

Signed-off-by: Radim Hrazdil <rhrazdil@redhat.com>

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**Release note**:

```release-note
Adds missing version prefix 'v' to manifests in released versions v0.41.0 and v0.42.0
```

### CODE REVIEW

**Comment by RamLavi** [comment]
PR description typo: `mathes`-->`matches`

File: manifests/cluster-network-addons/v0.42.0/operator.yaml [hunk 1]
```diff
@@ -121,7 +121,7 @@ apiVersion: apps/v1
 kind: Deployment
```
Comment (phoracek) [inline manifests/cluster-network-addons/v0.42.0/operator.yaml:121]: The directory structure and the version used in OLM is something we should not change. AFAIK OLM has issues with the `v` prefix.

We should keep it `v`-less everywhere except for git tags and image tags.

Comment (rhrazdil) [reply manifests/cluster-network-addons/v0.42.0/operator.yaml]: Thanks for explanation, updated

**Comment by rhrazdil** [comment]
https://github.com/kubevirt/cluster-network-addons-operator/issues/540 should be unrelated to this change
/test pull-e2e-cluster-network-addons-operator-kubemacpool-functests

**Review by phoracek — APPROVED** [review]
/lgtm
/approve

**Pull request merged**